text/3485-feature-documentation.md

```diff
@@ -0,0 +1,196 @@
+concerns about bloat, but this is further discussed in
+[future possibilities][future-possibilities].
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- Added complexity to Cargo.
+  - Exact implementation details do add test surface area
+  - A markdown parser is required to properly parse the `doc` field.
+- Docstrings can be lengthy, adding noise to `Cargo.toml`. This could
+  potentially be solved with the below mentioned `doc-file` key.
+- When rendering features in documentation, this RFC does not specify any way
+  for `rustdoc` to get the information it requires. This will require separate
+  design work.
+- Unlike with the [`document-features`](https://crates.io/crates/document-features)
+  crate there is no way to group features in into sections or have a
+  user-specified layout
+- Users cannot control features ordering in documentation since the TOML specification defines table keys as unordered.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- To avoid increasing the size of the registry index, this does not add `doc`
+  to a package's index entry.  This means a `.crate` file must be downloaded
+  and extracted to access the features.
+- Feature descriptions could be specified somewhere in Rust source files. This
+  has the downside of creating multiple sources of truth on features.
+- Cargo could parse doc comments in `Cargo.toml`, like the `document-features`
+  crate (linked below).
+
+  ```toml
+  # RFC proposal
+  foo = { enables = [], doc = "foo feature" }
+
+  # Alternative equivalent using doc comments
+  ## foo feature
+  foo = []
+  ```
+
+  This was decided against as part of this RFC because it would mean that
+  TOML-compliant parsers (including anything `serde`-based) would be
+  insufficient to extract all information in the manifest, requiring custom
+  deserialization of the fields via a format-preserving parser. This differs
+  from documentation in Rust source as the doc-comment behavior is described
+  specified within the grammar with parsers supporting extracting those
+  elements.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+- There is an existing crate that uses TOML comments to create a features table:
+  <https://docs.rs/document-features/latest/document_features/>
+- `docs.rs` displays a feature table, but it is fairly limited. If features
+  start with `_`, they are hidden from this table ([example](https://docs.rs/crate/regex/latest/features)).
+- `lib.rs` extracts feature documentation from `Cargo.toml` and source ([example](https://lib.rs/crates/regex/features))
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- Rather than being consistent with `rustdoc` and accepting markdown, should the
+  `doc` key be consistent with `package.description` and only support plain
+  text? This RFC proposes making this decision at time of implementation when the
+  challenges of supporting markdown are better understood.
+
```

Comment:
I found this confusing as the main body of the RFC makes it sound like a decision is made.  It might confuse people if they don't notice this here that they approve one thing and something different.

Reply:
Thanks for pointing this out, I added a note clarifying that this has not yet been decided (as you mention in https://github.com/rust-lang/rfcs/pull/3485#discussion_r1320674554)